forward_engineering/helpers/udtHelper.js

```diff
@@ -53,19 +53,17 @@ const isDefinitionTypeValidForAvroDefinition = definition => {
 }
 
 const resolveSchemaUdt = schema => {
-	const type = _.isString(schema) ? schema : schema.type;
+	const type = _.isString(schema) || _.isArray(schema) ? schema : schema.type;
 	if (isNativeType(type)) {
 		return schema;
 	}
 
 	if (_.isString(schema)) {
-		const typeFromUdt = getTypeFromUdt(schema);
-	
-		return typeFromUdt;
+		return getTypeFromUdt(schema);
 	}
 
 	if (_.isArray(type)) {
-		return { ...schema, type: type.map(resolveSchemaUdt) };
+		return { ...(_.isArray(schema) ? {} : schema), type: type.map(resolveSchemaUdt) };
```

Comment:
if schema has no properties - just array - we should not incorrectly destruct it with spread operator